src/commands/host_features.rs

```diff
@@ -5,7 +5,18 @@ use crate::{
 
 impl AdbTcpConnexion {
     /// Lists available ADB server features.
-    pub fn host_features(&mut self) -> Result<Vec<HostFeatures>> {
+    pub fn host_features<S: ToString + Clone>(
+        &mut self,
+        serial: Option<S>,
```

Comment:
We could take a `&Option<S>` instead

Reply:
Yeah, that was what I was wondering the most; but that'd require the entire API to change as you probably don't want some functions to take a reference for the serial parameter and some don't?
Requiring the Clone trait would keep the API (but restrict it's usage) so...

Am I understanding you correctly to make all functions taking a serial parameter as a reference, or should the 'host_features' function be a little special?